### PR TITLE
fix: invalidate confirmation interest on match request

### PR DESCRIPTION
**Pull request by dhenkel92**

### CODE REVIEW

File: common/match/pool.ts [hunk 1]
```diff
@@ -253,7 +253,7 @@ export const TEST_POOL = {
 const _pools = [
     {
         name: 'lern-fair-now',
-        confirmInterest: false,
+        confirmInterest: true,
```
Comment (Jonasdoubleyou) [inline common/match/pool.ts:256]: This needs to stay false

File: common/match/pool.ts [hunk 2]
```diff
@@ -386,9 +386,6 @@ export async function runMatching(poolName: string, apply: boolean, _toggles: st
     if (apply) {
         const startCommit = Date.now();
         for (const match of matches) {
-            if (pool.confirmInterest && match.pupil.openMatchRequestCount <= 1) {
-                await removeInterest(match.pupil);
-            }
```
Comment (Jonasdoubleyou) [inline common/match/pool.ts:391]: This can stay

File: common/match/create.ts [hunk 3]
```diff
@@ -50,6 +51,7 @@ export async function createMatch(pupil: Pupil, student: Student, pool: Concrete
     });
 
     await invalidateAllScreeningsOfPupil(pupil.id);
+    await removeInterest(pupil);
```
Comment (Jonasdoubleyou) [inline common/match/create.ts:54]: This is the actual fix

File: common/match/pool.ts (hunk 2 above)
Comment (dhenkel92) [reply common/match/pool.ts:391]: Is there a case where it's needed?

File: common/match/pool.ts (hunk 1 above)
Comment (dhenkel92) [reply common/match/pool.ts:256]: Then I misunderstood our conversation 😅 
I'll revert the change.

File: common/match/pool.ts (hunk 2 above)
Comment (Jonasdoubleyou) [reply common/match/pool.ts:391]: Ah nevermind, we call createMatch right after which now calls it anyways

File: common/match/pool.ts (hunk 1 above)
Comment (Jonasdoubleyou) [reply common/match/pool.ts:256]: Yeah we do not want to invite new users for interest confirmation, which this toggle controls

**Review by Jonasdoubleyou — APPROVED** [review]

**Pull request merged**